test/drivers/test_cqrrp.cc

```diff
@@ -0,0 +1,228 @@
+
+        auto m = all_data.row;
+        auto n = all_data.col;
+
+        CQRRP.call(m, n, all_data.A.data(), m, d_factor, all_data.tau.data(), all_data.J.data(), state);
+        all_data.rank = CQRRP.rank;
+        
+        RandLAPACK::util::upsize(all_data.rank * n, all_data.R);
+        lapack::lacpy(MatrixType::Upper, all_data.rank, n, all_data.A.data(), m, all_data.R.data(), all_data.rank);
+
+        
+        lapack::ungqr(m, n, n, all_data.A.data(), m, all_data.tau.data());
+        
+        lapack::lacpy(MatrixType::General, m, all_data.rank, all_data.A.data(), m, all_data.Q.data(), m);
+
+        printf("RANK AS RETURNED BY CQRRP %4ld\n", all_data.rank);
+
+        RandLAPACK::util::col_swap(m, n, n, all_data.A_cpy1.data(), m, all_data.J);
+        RandLAPACK::util::col_swap(m, n, n, all_data.A_cpy2.data(), m, all_data.J);
+
+        error_check(norm_A, all_data);
+    }
+};
+
+// Note: If Subprocess killed exception -> reload vscode
+TEST_F(TestCQRRP, CQRRP_blocked_full_rank_basic) {
+    int64_t m = 60;//5000;
+    int64_t n = 60;//2000;
+    int64_t k = 60;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 10;//500;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state = RandBLAS::RNGState();
+
+    CQRRPTestData<double> all_data(m, n, k);
+    RandLAPACK::CQRRP_blocked<double, r123::Philox4x32> CQRRP_blocked(true, tol, b_sz);
+    CQRRP_blocked.nnz = 2;
+    CQRRP_blocked.num_threads = 4;
+
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    //RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::polynomial);
+    //m_info.cond_num = 2;
+    //m_info.rank = k;
+    //m_info.exponent = 2.0;
+    RandLAPACK::gen::mat_gen<double, r123::Philox4x32>(m_info, all_data.A, state);
+
+    norm_and_copy_computational_helper<double, r123::Philox4x32>(norm_A, all_data);
+#if !defined(__APPLE__)
+    test_CQRRP_general<double, r123::Philox4x32, RandLAPACK::CQRRP_blocked<double, r123::Philox4x32>>(d_factor, norm_A, all_data, CQRRP_blocked, state);
+#endif
+}
+
+// Note: If Subprocess killed exception -> reload vscode
+TEST_F(TestCQRRP, CQRRP_blocked_full_rank_block_change) {
+    int64_t m = 32;//5000;
+    int64_t n = 32;//2000;
+    int64_t k = 32;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 7;//500;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state = RandBLAS::RNGState();
+
+    CQRRPTestData<double> all_data(m, n, k);
+    RandLAPACK::CQRRP_blocked<double, r123::Philox4x32> CQRRP_blocked(true, tol, b_sz);
+    CQRRP_blocked.nnz = 2;
+    CQRRP_blocked.num_threads = 4;
+
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    //RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::polynomial);
+    //m_info.cond_num = 2;
+    //m_info.rank = k;
+    //m_info.exponent = 2.0;
```

Comment:
Same as above. This should be removed unless there's a specific reason to keep it.

Reply:
Same as above